mosaic-runtime/src/main/kotlin/com/jakewharton/mosaic/Static.kt

```diff
@@ -0,0 +1,108 @@
+package com.jakewharton.mosaic
+
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ComposeNode
+import androidx.compose.runtime.DisallowComposableCalls
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.mutableStateListOf
+import androidx.compose.runtime.remember
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+
+/**
+ * Will render each value emitted by [items] as permanent output above the
+ * regular display.
+ */
+@Composable
+fun <T> Static(
+	items: Flow<T>,
+	content: @Composable (T) -> Unit,
+) {
+	class Item(val value: T, var rendered: Boolean)
+
+	// Keep list of items which have not yet been rendered.
+	val pending = remember { mutableStateListOf<Item>() }
+
+	LaunchedEffect(items) {
+		items.collect {
+			pending.add(Item(it, rendered = false))
+		}
+	}
+
+	Static(
+		postRender = {
+			// Remove any items which have been rendered.
+			pending.removeIf { it.rendered }
```

Comment:
This isn't super efficient but it's probably fine. One thing to note is that the `true`s are contiguous starting at index 0 so we could find the first `false` index and then do `subList(0, firstFalse).clear()` for an operation that is likely the most optimal.

Reply:
That's a good idea. I always forget that modifications on subList are reflected in the full list. Also worth noting that just `pending.clear()` worked in all of the test samples I created but I wasn't convinced that would work in all cases so I added the wrapper `Item` class to be safe.